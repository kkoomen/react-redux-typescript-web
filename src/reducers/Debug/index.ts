import { debugReducer, initialState as debugState } from './reducer';
import * as debugActions from './actions';

export { debugActions };
export * from './types';
export {
  debugReducer,
  debugState,
};
