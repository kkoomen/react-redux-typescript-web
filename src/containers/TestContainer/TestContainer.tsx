import React, { Component } from 'react';
import { ConnectedReduxProps } from '../../reducers';

interface Props {};

interface State {};

type TestContainerProps = Props & ConnectedReduxProps;

export class TestContainer extends Component<TestContainerProps, State> {
  public state = {};

  public render() {
    return (
      <div className="TestContainer" />
    );
  }
}

export default TestContainer;
