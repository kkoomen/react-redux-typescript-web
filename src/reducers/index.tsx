import { combineReducers, Dispatch, Action, AnyAction } from 'redux'
import { RouterState } from 'connected-react-router';
import { persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage';

import debugReducer from './Debug/reducer';
import { DebugState } from './Debug/types';

export interface AppState {
  debug: DebugState,
};

// Additional props for connected React components. This prop is passed by default with `connect()`
export interface ConnectedReduxProps<A extends Action = AnyAction> {
  dispatch: Dispatch<A>
}

const rootReducer = combineReducers<AppState>({
  debug: debugReducer,
});

const localStoragePersistConfig = {
  key: process.env.APP_NAME,
  storage,
  // Only keys specified in the 'whitelist' property will be persisted.
  whitelist: ['debug'],
};

export default persistReducer(localStoragePersistConfig, rootReducer);
