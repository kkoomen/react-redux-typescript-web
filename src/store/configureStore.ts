import {
  Store,
  createStore,
  applyMiddleware,
  compose,
  Middleware,
  StoreEnhancer,
} from 'redux';
import { routerMiddleware } from 'connected-react-router'
import { createLogger } from 'redux-logger';
import { History } from 'history'
import thunk from 'redux-thunk';
import { __DEBUG__ } from '../config/constants';
import { rootReducer, AppState } from '../reducers';

// Compose with the devtools extension when in a non-production environment.
const composeEnhancers = (
  __DEBUG__
  && window
  && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
) || compose;

export function configureStore(
  history: History,
  initialState: object
): Store<AppState> {
  const enhancers: StoreEnhancer[] = [];
  const middlewares: Middleware[] = [
    thunk,
    routerMiddleware(history),
  ];

  if (__DEBUG__) {
    console.log('[setup] âœ“ Enabling state logger');
    const loggerMiddleware = createLogger({
      level: 'info',
      collapsed: true,
      stateTransformer: (state) => state,
      predicate: (getState, action) => {
        const state = getState();
        return state.debug.logs.enabled;
      },
    });
    middlewares.push(loggerMiddleware);
  }

  const composedEnhancers = composeEnhancers(
    applyMiddleware(...middlewares),
    ...enhancers
  );

  const store = createStore(
    rootReducer(history),
    initialState,
    composedEnhancers
  );

  return store;
}
