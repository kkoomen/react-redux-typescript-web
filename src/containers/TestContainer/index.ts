import { connect } from 'react-redux';
import { TestContainer as TestContainerContainer } from './TestContainer';
import { AppState } from '../../reducers';

function mapStateToProps(state: AppState) {
  return {};
}

export const TestContainer = connect(mapStateToProps)(TestContainerContainer);
