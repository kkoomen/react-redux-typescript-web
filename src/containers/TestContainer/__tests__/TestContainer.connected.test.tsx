import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import { testContainerState } from '../../../reducers/TestContainer';
import { TestContainer } from '../';

const mockStore = configureStore();
const initialStoreState = {};

let store: any;
beforeEach(() => {
  store = mockStore({
    testContainer: testContainerState,
  });
});

describe('TestContainer container', () => {
  it('should match its snapshot by default', () => {
    const element = mount(
      <Provider store={store}>
        <TestContainer />
      </Provider>
    );
    expect(toJson(element)).toMatchSnapshot();
  });
});
