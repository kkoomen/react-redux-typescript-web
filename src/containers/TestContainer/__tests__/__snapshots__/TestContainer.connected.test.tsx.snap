// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestContainer container should match its snapshot by default 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(TestContainer)>
    <TestContainer
      dispatch={[Function]}
    >
      <div
        className="TestContainer"
      />
    </TestContainer>
  </Connect(TestContainer)>
</Provider>
`;
